### PR TITLE
SAK-30813 Update VeriCite integration for new API and Content Review

### PR DESCRIPTION
**Pull request by baholladay**

### CODE REVIEW

File: content-review/impl/vericite/pom.xml [hunk 1]
```diff
@@ -42,5 +52,62 @@
...
+                        <artifactId>jersey-multipart</artifactId>
+                        <version>${jersey-version}</version>
+                </dependency>
+                <dependency>
+                        <groupId>io.swagger</groupId>
+                        <artifactId>swagger-annotations</artifactId>
+                        <version>${swagger-annotations-version}</version>
+                </dependency> 
+                <!-- JSON processing: jackson -->
+                <dependency>
```
Comment (ern) [inline content-review/impl/vericite/pom.xml]: Jackson is already in shared just need to add dependency with groupId and artifactId.

File: content-review/impl/vericite/pom.xml [hunk 2]
```diff
@@ -42,5 +52,62 @@
...
+                </dependency>
+                <dependency>
+                        <groupId>com.fasterxml.jackson.datatype</groupId>
+                        <artifactId>jackson-datatype-joda</artifactId>
+                        <version>2.1.5</version>
+                </dependency>
+                <dependency>
+                        <groupId>joda-time</groupId>
+                        <artifactId>joda-time</artifactId>
+                        <version>${jodatime-version}</version>
```
Comment (ern) [inline content-review/impl/vericite/pom.xml]: joda-time version is specifiied in master probably should look to using it ${joda.time.version}

File: content-review/impl/vericite/src/main/java/org/sakaiproject/contentreview/vericite/ContentReviewServiceVeriCite.java [hunk 3]
```diff
@@ -317,103 +341,102 @@ private String getAccessUrl(String contentId, String assignmentRef, String userI
...
-					if(results != null){
-						//check for error message:
-						String errorMsg = getErrorMessage(results);
-						if(errorMsg == null){
-							token = results.getString(PARAM_TOKEN);
+			String cacheKey = context + ":" + assignmentId + ":" + userId;
+			//first check if cache already has the URL for this contentId and user
+			if(userUrlCache.containsKey(cacheKey)){
+				Object cacheObj = userUrlCache.get(cacheKey);
+				if(cacheObj != null && cacheObj instanceof Map){
```
Comment (ern) [inline content-review/impl/vericite/src/main/java/org/sakaiproject/contentreview/vericite/ContentReviewServiceVeriCite.java]: Does it make sense to extract this into a concrete object?

File: content-review/impl/vericite/src/main/java/org/sakaiproject/contentreview/vericite/ContentReviewServiceVeriCite.java [hunk 4]
```diff
@@ -541,101 +616,240 @@ public boolean isSiteAcceptable(Site arg0) {
...
-
-	public void removeFromQueue(String arg0) {
-		// TODO Auto-generated method stub
-		
+	
+	public void queueContent(String userId, String siteId, String assignmentReference, List<ContentResource> content) throws QueueException{
+		crqs.queueContent(getProviderId(), userId, siteId, assignmentReference, content);
+	}
+
+//	private void queue(final String userId, String siteId, final String assignmentReference, final List<FileSubmission> fileSubmissions){
```
Comment (ern) [inline content-review/impl/vericite/src/main/java/org/sakaiproject/contentreview/vericite/ContentReviewServiceVeriCite.java]: should just remove this

File: content-review/impl/vericite/pom.xml (hunk 1 above)
Comment (baholladay) [inline content-review/impl/vericite/pom.xml]: Updated and will put in new PR

File: content-review/impl/vericite/pom.xml (hunk 2 above)
Comment (baholladay) [inline content-review/impl/vericite/pom.xml]: updated and will put in new PR

File: content-review/impl/vericite/src/main/java/org/sakaiproject/contentreview/vericite/ContentReviewServiceVeriCite.java (hunk 3 above)
Comment (baholladay) [inline content-review/impl/vericite/src/main/java/org/sakaiproject/contentreview/vericite/ContentReviewServiceVeriCite.java]: Thanks

**Pull request merged**